test/expect/pure/formats/query/query.ml

```diff
@@ -5,22 +5,30 @@
 
 
 
-let query name string =
-  Dream.query (Dream.request ~target:("/?" ^ string) "") name
+let query name string ~filter_query =
+  Dream.query (Dream.request ~target:("/?" ^ string) "") name ~filter_query ()
   |> function
     | Some value -> Printf.printf "%S\n" value
     | None -> print_endline "None"
 
 let%expect_test _ =
-  query "a" "a=b";
-  query "" "a=b";
-  query "" "";
-  query "a" "";
-  query "a" "a=";
-  query "" "=a";
-  query "a" "a=b&a=c";
-  query "c" "a=b&c=d";
-  query "a b" "a+b=c";
+  query ~filter_query:false "a" "a=b";
+  query ~filter_query:false "" "a=b";
+  query ~filter_query:false "" "";
+  query ~filter_query:false "a" "";
+  query ~filter_query:false "a" "a=";
+  query ~filter_query:false "" "=a";
+  query ~filter_query:false "a" "a=b&a=c";
+  query ~filter_query:false "c" "a=b&c=d";
+  query ~filter_query:false "a b" "a+b=c";
+  query ~filter_query:false "a" "a";
+  query ~filter_query:true "a" "a";
+  query ~filter_query:true "a" "a ";
+  query ~filter_query:true "a" "a  ";
+  query ~filter_query:true "a" "a=";
+  query ~filter_query:true "a" "a= ";
+  query ~filter_query:true "a" "a=  ";
+  query ~filter_query:true "a" "a= &a=b";
```

Comment:
That one currently gives `None`. Is that what we want?

Reply:
Yes. [`Dream.query`](https://aantron.github.io/dream/#val-query):

> First query parameter with the given name.

[`Dream.queries`](https://aantron.github.io/dream/#val-queries) can be used to get the result `[" ", "b"]` here.